### PR TITLE
highlight changes in CompareRunView

### PR DESCRIPTION
**Pull request by gabrielbretschner**
## What changes are proposed in this pull request?

This PR introduces highlighting of different parameter values when comparing runs. This is achieved by adding a class `row-changed` to the parameter row and styling it with CSS. I just chose a color myself as I was unsure which colors should be used. 

## How is this patch tested?

Tested in local installation

### Is this a user-facing change?

- [x] Yes. Give a description of this change to be included in the release notes for MLflow users.

When comparing runs, parameters are highlighted where the runs differ.

### What component(s) does this PR affect?

- [x] UI
- [ ] CLI
- [ ] API
- [ ] REST-API
- [ ] Examples
- [ ] Docs
- [ ] Tracking
- [ ] Projects
- [ ] Artifacts
- [ ] Models
- [ ] Scoring
- [ ] Serving
- [ ] R
- [ ] Java
- [ ] Python

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [x] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: mlflow/server/js/src/components/CompareRunView.js [hunk 1]
```diff
@@ -153,7 +156,21 @@ class CompareRunView extends Component {
     });
 
     return keys.map(k => {
-      return <tr key={k}>
+      let row_class = undefined;
+      if (highlightChanges) {
+        const all_equal = data[k].reduce((acc, currentValue) => {
```
Comment (AveshCSingh) [inline mlflow/server/js/src/components/CompareRunView.js:161]:
```suggestion
        data[k].every(x => x == data[k][0])
```

I think the reduce is unnecessarily confusing. Could you just compare each value to the 0'th value?

**Review by AveshCSingh — CHANGES_REQUESTED** [review]
Thank you for the PR, @gabrielbretschner. This looks good to me, modulo one small request inline.

@gioa, do you know if MLflow has a predefined color palette?

Screen Shot 2020-04-03 at 4 22 37 PM

**Comment by gioa** [comment]
> @gioa, do you know if MLflow has a predefined color palette?

@gabrielbretschner Thank you for the work! The orange looks too prominent for me. Maybe use a light yellow for the background? See the attached mock, I use a rgba(249, 237, 190, 0.5) and it passes the [WCAG color contrast](https://developer.mozilla.org/en-US/docs/Web/Accessibility/Understanding_WCAG/Perceivable/Color_contrast) guideline. 

Diff

Another related question, what do you think of adding a switch to show different parameters only? Would that be useful for people to better compare runs?

File: mlflow/server/js/src/components/CompareRunView.js (hunk 1 above)
Comment (gabrielbretschner) [reply mlflow/server/js/src/components/CompareRunView.js]: Good point. I simplified the reduce

**Comment by gabrielbretschner** [comment]
> > @gioa, do you know if MLflow has a predefined color palette?
> 
> @gabrielbretschner Thank you for the work! The orange looks too prominent for me. Maybe use a light yellow for the background? See the attached mock, I use a rgba(249, 237, 190, 0.5) and it passes the [WCAG color contrast](https://developer.mozilla.org/en-US/docs/Web/Accessibility/Understanding_WCAG/Perceivable/Color_contrast) guideline.
> 
> Diff

Thank you for your suggestion. That looks much better.

> Another related question, what do you think of adding a switch to show different parameters only? Would that be useful for people to better compare runs?

That would be great, though I don't know how to implement this (I'm not a webdev...)

**Review by AveshCSingh — COMMENTED** [review]
LGTM. I don't believe the JS test failure is related, so I've retriggered the test suite. Once it passes, you can submit.

Thanks @gabrielbretschner!

**Review by AveshCSingh — APPROVED** [review]

**Pull request merged**